third_party/2and3/jinja2/utils.pyi

```diff
@@ -1,26 +1,51 @@
-from typing import Any, Callable, Iterable, Optional
+import sys
+from typing import Any, Callable, IO, Iterable, Optional, Text, TypeVar, Union
 
 from markupsafe import Markup as Markup, escape as escape, soft_unicode as soft_unicode
 
 missing: Any
 internal_code: Any
 concat: Any
 
-def contextfunction(f): ...
-def evalcontextfunction(f): ...
-def environmentfunction(f): ...
-def internalcode(f): ...
-def is_undefined(obj): ...
+if sys.version_info >= (3, 6):
+    from builtins import _PathLike
+    _PathType = Union[bytes, Text, _PathLike]
+else:
+    _PathType = Union[bytes, Text]
+if sys.version_info >= (3,):
+    text_type = str
+else:
+    text_type = unicode
+
+_T = TypeVar('_T')
+
+class ContextFunction():
+    contextfunction: bool
+    def __call__(self, *args: Any, **kwargs: Any) -> _T: ...
+
+class EvalContextFunction():
+    evalcontextfunction: bool
+    def __call__(self, *args: Any, **kwargs: Any) -> _T: ...
+
+class EnvironmentFunction():
+    environmentfunction: bool
+    def __call__(self, *args: Any, **kwargs: Any) -> _T: ...
+
+def contextfunction(f: Callable[..., _T]) -> ContextFunction: ...
+def evalcontextfunction(f: Callable[..., _T]) -> EvalContextFunction: ...
+def environmentfunction(f: Callable[..., _T]) -> EnvironmentFunction: ...
+def internalcode(f: Callable[..., _T]) -> Callable[..., _T]: ...
+def is_undefined(obj: object) -> bool: ...
 def select_autoescape(enabled_extensions: Iterable[str] = ..., disabled_extensions: Iterable[str] = ..., default_for_string: bool = ..., default: bool = ...) -> Callable[[str], bool]: ...
-def consume(iterable): ...
-def clear_caches(): ...
-def import_string(import_name, silent: bool = ...): ...
-def open_if_exists(filename, mode: str = ...): ...
-def object_type_repr(obj): ...
-def pformat(obj, verbose: bool = ...): ...
-def urlize(text, trim_url_limit: Optional[Any] = ..., nofollow: bool = ..., target: Optional[Any] = ...): ...
-def generate_lorem_ipsum(n: int = ..., html: bool = ..., min: int = ..., max: int = ...): ...
-def unicode_urlencode(obj, charset: str = ..., for_qs: bool = ...): ...
+def consume(iterable: Iterable) -> None: ...
+def clear_caches() -> None: ...
+def import_string(import_name: str, silent: bool = ...): ...
+def open_if_exists(filename: _PathType, mode: str = ...) -> IO[Any]: ...
```

Comment:
```suggestion
def open_if_exists(filename: _PathType, mode: str = ...) -> Optional[IO[Any]]: ...
```
Could use overloads on literal values of `mode` to get a more precise return type, but that may not be worth it.